Added support for outputting skipped tests to junit

Andne opened the pull request:
Fixes #462.  It's not a very good solution, just checks what character the progress indicator is, but it works.  It looks to me like there is some unneeded coupling going on between the TestOutput logic and the UTestShell class since it knows the character that gets printed.  I think that it should probably just know it's status and let the TestOutput class decide which character that means gets printed, or in the case of JUnit Output, what to put into the file.

basvodde commented on src/CppUTest/JUnitTestOutput.cpp:
```diff
@@ -135,6 +136,9 @@ void JUnitTestOutput::printCurrentTestStarted(const UtestShell& test)
         impl_->results_.tail_ = impl_->results_.tail_->next_;
     }
     impl_->results_.tail_->name_ = test.getName();
+    if (*(test.getProgressIndicator()) == '!') {
```
This is kind of an ugly way of discovering ignored. Would it be better to add an willRun method or so to the uTest class?

Andne replied:
Probably, I was at the time just trying to make it work.  Had thoughts of adding a status to Utest that would indicate not run, passed, failed or skipped, but never tried to do anything with it.

basvodde replied:
Could you change it still? This way the JUnitOutput would break down if we change the ignored indicator on the command line. I'd say that is a weird side-effect :)

basvodde commented on src/CppUTest/Utest.cpp:
```diff
@@ -596,6 +596,11 @@ IgnoredUtestShell::IgnoredUtestShell()
 {
 }
 
+IgnoredUtestShell::IgnoredUtestShell(const char* groupName, const char* testName, const char* fileName, int lineNumber) :
```
Wasn't this there yet?

Andne replied:
I got linker errors about it missing until I added it, so I guess it wasn't.

basvodde replied:
Interesting.

Andne commented:
I changed it to depend on a willRun function.  I also removed the progress indicator from UtestShell and TestResult since it seemed to be just a function of the console test output and not actually part of the test itself, so having one outputter being told what to print and the other knowing it based on data didn't seem quite right to me.

basvodde commented:
Ok, interesting.
